### PR TITLE
enable all available  tests for zynq7000-qemu (waiting for QuickRunner)

### PR DESCRIPTION
**Pull request by damianloew**
## Description

- trunner: restore psh tests on zynq7000-qemu
- enable running unit tests on zynq7000-qemu
- psh: temporarily disable test-kill for zynq7000-qemu

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [x] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [ ] Tested by hand on: (list targets here).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [x] This PR needs additional PRs to work (https://github.com/phoenix-rtos/phoenix-rtos-project/pull/449).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -14,7 +14,7 @@ on:
 
 jobs:
   call-ci:
-    uses: phoenix-rtos/phoenix-rtos-project/.github/workflows/ci-submodule.yml@master
+    uses: phoenix-rtos/phoenix-rtos-project/.github/workflows/ci-submodule.yml@damianloew/zynq7000_ci_ports
```
Comment (jsarzy) [inline .github/workflows/ci.yml:17]: Marking it as a merge blocker.

File: libc/test.yaml [hunk 2]
```diff
@@ -4,5 +4,6 @@ test:
         - name: unit
           exec: test-libc
           #FIXME: symlink tests will fail on non-rootfs targets
+          # TODO: enable the test for armv7a9-zynq7000-qemu after resolving #445 issue
```
Comment (jsarzy) [inline libc/test.yaml:7]: Maybe paste a link to the issue on a new line

File: trunner/testcase.py [hunk 3]
```diff
@@ -89,11 +91,11 @@ def sysexec(self, proc):
     def exec_test(self, proc):
         try:
             # Wait for a prompt
-            # On armv7a9-zynq7000-qemu jffs initialization may take to ~20s, that's why it's set to 25
+            # On armv7a9-zynq7000-qemu jffs initialization may take to ~25s, that's why it's set to 30
```
Comment (jsarzy) [inline trunner/testcase.py:94]: We are moving in wrong direction :face_with_head_bandage:

File: trunner/testcase.py [hunk 4]
```diff
@@ -308,13 +313,18 @@ class TestCaseFactory:
     def create(test):
         status = TestCase.SKIPPED if test['ignore'] else TestCase.FAILED
         use_sysexec = test['target'] in SYSEXEC_TARGETS
+        if test['target'] in ASH_TARGETS:
+            prompt = ASH_PROMPT
+        else:
+            prompt = DEFAULT_PROMPT
```
Comment (jsarzy) [inline trunner/testcase.py:319]: I don't know if it's a good way to add target dependencies here. I noticed that when a new target comes then the list of arguments to the `TestCaseUnit` also grows. Maybe we should create `TestCaseArgs` or `TestCaseConfig` and make this ifology in the constructor? Then we could pass one argument to `TestCaseUnit` and use it there (maybe unpacking it or using property feature). Then we could also eliminate some ifs from `TestCaseUnit` methods.

File: trunner/config.py [hunk 5]
```diff
@@ -40,15 +40,21 @@ def resolve_phrtos_dir() -> Path:
...
 
-SYSEXEC_TARGETS = ['armv7a9-zynq7000-qemu', 'armv7m7-imxrt106x-evk', 'armv7m7-imxrt117x-evk', 'riscv64-generic-qemu']
+SYSEXEC_TARGETS = ['armv7m7-imxrt106x-evk', 'armv7m7-imxrt117x-evk', 'riscv64-generic-qemu']
 
 QEMU_TARGETS = ['armv7a9-zynq7000-qemu', 'ia32-generic-qemu', 'riscv64-generic-qemu']
 
+# Targets, where busybox ash starts by default
+ASH_TARGETS = ['armv7a9-zynq7000-qemu']
+
+DEFAULT_PROMPT = "(psh)% "
```
Comment (jsarzy) [inline trunner/config.py:55]: Maybe define `(psh)% ` as `PSH_PROMPT` and then `DEFAULT_PROMPT = PSH_PROMPT`

**Review by jsarzy — CHANGES_REQUESTED** [review]
some things to discuss, can't be merged before https://github.com/phoenix-rtos/phoenix-rtos-project/pull/449

File: trunner/testcase.py (hunk 3 above)
Comment (damianloew) [reply trunner/testcase.py:94]: The ash start-up takes some time, that's why it's increased here. This problem is known and it's planned to resolve it, so I've created issue (https://github.com/phoenix-rtos/phoenix-rtos-project/issues/450) for that and mentioned it in the comment.

File: trunner/config.py (hunk 5 above)
Comment (damianloew) [reply trunner/config.py:55]: Yeah, it may be better

File: trunner/testcase.py (hunk 4 above)
Comment (damianloew) [reply trunner/testcase.py]: It's done in the last force push

**Comment by damianloew** [comment]
Moved to https://github.com/phoenix-rtos/phoenix-rtos-tests/pull/161

**Pull request closed without merging**